src/axom/sina/core/CurveSet.hpp

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2017-2024, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef SINA_CURVESET_HPP
+#define SINA_CURVESET_HPP
+
+/*!
+ ******************************************************************************
+ *
+ * \file CurveSet.hpp
+ *
+ * \brief   Implementation file for Sina CurveSet class
+ *
+ * \sa Curve.hpp
+ *
+ ******************************************************************************
+ */
+
+#include <string>
+#include <unordered_map>
+
+#include "conduit.hpp"
+
+#include "axom/sina/core/Curve.hpp"
+
+namespace axom
+{
+namespace sina
+{
+
+/**
+ * \brief A CurveSet represents an entry in a record's "curve_set".
+ *
+ * A CurveSet consist of a set of independent and dependent curves. Each curve
+ * is a list of numbers along with optional units and tags.
+ *
+ * \sa Record
+ * \sa Curve
+ */
+class CurveSet
+{
+public:
+  /**
+     * An unordered map of Curve objects.
+     */
+  using CurveMap = std::unordered_map<std::string, Curve>;
```

Comment:
The use case here is similar to sidre's `MapCollection`  and `ItemCollection` classes, which are at the foundation of `sidre::Group`. 

In the future, we might want to consider generalizing sidre's collections, move them to `core` and using them here.

See: 
* https://github.com/LLNL/axom/blob/develop/src/axom/sidre/core/MapCollection.hpp
* https://github.com/LLNL/axom/blob/develop/src/axom/sidre/core/ItemCollection.hpp

Reply:
@kennyweiss I made an issue for this: https://github.com/LLNL/axom/issues/1397